ddl/ddl_worker.go

```diff
@@ -185,28 +185,62 @@ func buildJobDependence(t *meta.Meta, curJob *model.Job) error {
 	return nil
 }
 
-// addDDLJob gets a global job ID and puts the DDL job in the DDL queue.
-func (d *ddl) addDDLJob(ctx sessionctx.Context, job *model.Job) error {
+func (d *ddl) limitDDLJobs() {
+	for {
+		tasks := make([]*limitJobTask, 0, batchAddingJobs)
+		select {
+		case task := <-d.limitJobCh:
```

Comment:
I do not quite understand how this limits the task rate.
By let multiple worker conflict on the same channel and increase the runtime lock spin?

Reply:
Write conflicts occur when multiple DDL jobs are written to the DDL job list at the same time.
We use this channel to reduce such conflicts.